lib/pages/chat/events/message_video_download_content.dart

```diff
@@ -0,0 +1,119 @@
+import 'package:fluffychat/pages/chat/events/event_video_player.dart';
+import 'package:fluffychat/pages/chat/events/message_content_style.dart';
+import 'package:fluffychat/presentation/mixins/handle_download_file_from_queue_in_mobile_mixin.dart';
+import 'package:fluffychat/presentation/mixins/play_video_action_mixin.dart';
+import 'package:fluffychat/presentation/model/chat/downloading_state_presentation_model.dart';
+import 'package:flutter/material.dart';
+import 'package:linagora_design_flutter/colors/linagora_ref_colors.dart';
+import 'package:matrix/matrix.dart';
+
+class MessageVideoDownloadContent extends StatefulWidget {
+  const MessageVideoDownloadContent({
+    super.key,
+    required this.event,
+    required this.width,
+    required this.height,
+  });
+
+  final Event event;
+
+  final double width;
+
+  final double height;
+
+  @override
+  State<StatefulWidget> createState() => _MessageVideoDownloadContentState();
+}
+
+class _MessageVideoDownloadContentState
+    extends State<MessageVideoDownloadContent>
+    with HandleDownloadFileFromQueueInMobileMixin, PlayVideoActionMixin {
+  @override
+  void initState() {
+    super.initState();
+    checkDownloadFileState(widget.event);
```

Comment:
you can put initState and dispose in `HandleDownloadFileFromQueueInMobileMixin` using 
```dart
mixin HandleDownloadFileFromQueueInMobileMixin<T extends StatefulWidget> on State<T> {
  @override
  void initState(){
    super.initState();
    // method call
  }

  @override
  void dispose(){
    // dispose whatever you need to
    super.dispose();
  }
}
```
This way, if someone needs to use this mixin, the chances to avoid calling `initState` and `dispose` are null

Reply:
yeah we can put `initState` and `dispose` too

Reply:
But `on State<T>` will narrow the usability of the mixin i think

Reply:
why ?